cpp/benchmarks/common/generate_input.hpp

```diff
@@ -403,3 +403,37 @@ std::unique_ptr<cudf::table> create_random_table(std::vector<cudf::type_id> cons
                                                  row_count num_rows,
                                                  data_profile const& data_params = data_profile{},
                                                  unsigned seed                   = 1);
+
+/**
+ * @brief Generate sequence columns starting with value 0 in first row and increasing by 1 in
+ * subsequent rows.
+ *
+ * If the number of passed types is smaller than the number of requested columns, the types
+ * will be repeated cyclically to fill the number of requested columns.
```

Comment:
This seems like astonishing behavior. I would expect the caller create the exact list of the number and types of column they want.

Reply:
I understand _why_, I just don't feel like it's an important enough use case to justify the astonishing behavior.

Reply:
@karthikeyann I agree. I think the action I would take is to call `repeat_dtypes` and pass its result into the `create_sequence_table` function wherever the number of input dtypes does not match the number of columns.

I might also rename `repeat_dtypes` to `cycle_dtypes`. "Cycle" is less ambiguous than "repeat" because it makes clear that you don't repeat the items individually, like `{0, 0, 1, 1, 2, 2}` vs. `{0, 1, 2, 0, 1, 2}`. This naming convention also aligns with Python's itertools.
https://docs.python.org/3/library/itertools.html#itertools.cycle
https://docs.python.org/3/library/itertools.html#itertools.repeat

Reply:
I agree. Cycle is more apt name. 
`itertools.repeat` repeats the entire object, not iterable. But I got the point.

Reply:
Oh, good catch. You're correct about `itertools.repeat`.

Reply:
moved `cycle_dtypes` outside. This simplifies a lot of code.